merkle_tree/src/main.rs

```diff
@@ -0,0 +1,432 @@
+use sha256::digest;
+
+struct Merkle_Tree {
+    tree: Vec<String>,
+    leaves: usize,
+    og_leaves: Vec<String>,
+}
+/* */
+impl Merkle_Tree {
+    /// Builds the tree concatenating the diferent levels into one array.
+    /// The final array tree goes from root to leaves -> [root....branch_n, branch_n+1....leaf].
+    fn new(data_vector: Vec<&str>) -> Self {
+        let string_data_vec: Vec<String> = data_vector.iter().map(|s| s.to_string()).collect();
```

Comment:
Nit-pick: to_string() can be a bit confusing. I'd use:[ to_owned()](https://doc.rust-lang.org/std/borrow/trait.ToOwned.html#tymethod.to_owned).